libdirections-session/src/test/java/com/mapbox/navigation/directions/session/MapboxDirectionsSessionTest.kt

```diff
@@ -17,28 +19,49 @@ class MapboxDirectionsSessionTest {
 
     private val router: Router = mockk(relaxUnitFun = true)
     private val origin: Point = mockk(relaxUnitFun = true)
+    private val destination: Point = mockk(relaxUnitFun = true)
     private val waypoints: List<Point> = mockk(relaxUnitFun = true)
     private val observer: DirectionsSession.RouteObserver = mockk(relaxUnitFun = true)
-    private lateinit var routeCallback: ((route: Route) -> Unit)
+    private lateinit var callback: Router.Callback
     private val route: Route = mockk(relaxUnitFun = true)
 
     @Before
     fun setUp() {
-        session = MapboxDirectionsSession(router)
-        session.setOrigin(origin)
-        session.setWaypoints(waypoints)
+        val listener = slot<Router.Callback>()
+        every { router.getRoute(origin, waypoints, destination, capture(listener)) } answers {
+            callback = listener.captured
+        }
+        session = MapboxDirectionsSession(router, origin, waypoints, destination)
     }
 
     @Test
     fun initialRouteResponse() {
         assertNull(session.currentRoute)
-        routeCallback.invoke(route)
+        session.registerRouteObserver(observer)
+        session.setWaypoints(waypoints)
```

Comment:
This is not testing the initial request anymore. It's testing a request that's sent after the waypoints are reset and actually catches that the initial request is not being sent (after commenting `session.setWaypoints(waypoints)` out).

You can extract the waypoint logic test to a different one and leave this one without it. You can then fix the failing initial request test by moving `origin` and `waypoints` to the constructor declaration. That's because the class' `init` block that makes the initial request is executed before assignments.

Reply:
fixed

Reply:
changed API